src/main/kotlin/io/github/moulberry/notenoughupdates/miscgui/DynamicLightItemsEditor.kt

```diff
@@ -0,0 +1,232 @@
+            tooltip.add("§eitems that will emit dynamic light when held. Simply hold the")
+            tooltip.add("§eitem you wish to add, then open this menu again and click")
+            tooltip.add("§e'Add Held Item', now if you have OptiFine installed and the")
+            tooltip.add("§edynamic lights option enabled, the added items will emit light!")
+            tooltip.add("")
+            tooltip.add("§eTo remove an item, click the item in this menu and click")
+            tooltip.add("§ethe 'Remove Item' button in the bottom right.")
+            Utils.drawHoveringText(tooltip, mouseX, mouseY, width, height, -1)
+        }
+
+        super.drawScreen(mouseX, mouseY, partialTicks)
+    }
+
+    override fun mouseClicked(mouseX: Int, mouseY: Int, mouseButton: Int) {
+        // Add Held Item button
+        if (mouseX >= guiLeft + 15 &&
+            mouseX <= guiLeft + 103 &&
+            mouseY >= (guiTop + ySize - 32) &&
+            mouseY <= (guiTop + ySize - 12)) {
+
+            val heldItem = Minecraft.getMinecraft().thePlayer.heldItem
+
+            if (heldItem == null) {
+                Utils.addChatMessage("§c[NEU] You can't add your hand to the list of dynamic light items.")
+                return
+            }
+
+            val internalName = resolveInternalName(heldItem)
+            if (internalName == null) {
+                Utils.addChatMessage("§c[NEU] Couldn't resolve an internal name for this item!")
+                return
+            }
+            NotEnoughUpdates.INSTANCE.config.hidden.dynamicLightItems.add(internalName)
+        }
+
+        // Remove Item button
+        if (mouseX >= guiLeft + 114 &&
+            mouseX <= guiLeft + 202 &&
+            mouseY >= guiTop + ySize - 32 &&
+            mouseY <= guiTop + ySize - 12 &&
+            itemSelected != null) {
+            NotEnoughUpdates.INSTANCE.config.hidden.dynamicLightItems.remove(itemSelected)
+            itemSelected = null
+        }
+
+        if (stackToRender != null) {
+            itemSelected = stackToRender
+        }
+
+        super.mouseClicked(mouseX, mouseY, mouseButton)
+    }
+
+    companion object {
+        fun resolveItemStack(internalName: String): ItemStack? {
+            var itemStack = NotEnoughUpdates.INSTANCE.manager
+                .createItemResolutionQuery()
+                .withKnownInternalName(internalName)
+                .resolveToItemStack()
+            if (itemStack == null) {
+                // Try resolve the item stack through forge
+                itemStack = GameRegistry.makeItemStack(internalName, 0, 1, null)
+                if (itemStack == null) {
+                    Utils.addChatMessage("§c[NEU] Couldn't resolve the ItemStack for $internalName")
+                    return null
+                }
+            }
+
+            return itemStack
+        }
+
+        fun resolveInternalName(itemStack: ItemStack): String? {
```

Comment:
btw you can't use `@JvmStatic` to avoid having to use `.Companion` in java sources.